sdk/cosmos/azure_data_cosmos/src/pipeline/mod.rs

```diff
@@ -105,7 +108,7 @@ impl CosmosPipeline {
                     req.insert_header(constants::CONTINUATION, continuation);
                 }
 
-                let resp = pipeline.send(&ctx, &mut req).await?;
+                let resp = pipeline.send(&ctx, &mut req).await?.success().await?;
```

Comment:
@analogrelay would it be better to implement this on `IntoFuture<Item = Response<T, F>>` such that we're not having to await twice? Granted, customers aren't having to type that themselves, but I imagine this compiles into separate `poll_next` calls against. If it's optimized away, fine. Just seems overkill this way.